ensime_shared/ensime.py

```diff
@@ -279,11 +336,15 @@ def au_buf_leave(self, client, filename):
 
     @execute_with_client()
     def au_cursor_hold(self, client, filename):
-        client.on_cursor_hold(filename)
+        """Handler for CursorHold autocommand event."""
+        self.watchdog_client_server_connection(client)
+        client.unqueue_and_display(filename)
+        client.editor.cursorhold()
```

Comment:
I've been using this PR for the last days. It works well, though I've discovered that vim is slower to respond to keystrokes and, sometimes, events are fired by no reason. For instance, when typing code, the UI gets blocked. I infer this happens because ensime-vim is querying the server and waiting for its response. I don't know if this is a consequence of what @ktonga points out.